source/LifecycleMethod.h

```diff
@@ -135,6 +135,9 @@ class LifeCycleMethodGraph {
 
   const LifecycleGraphNode* get_node(const std::string& node_name) const;
   bool operator==(const LifeCycleMethodGraph& other) const;
+  std::unordered_map<std::string, LifecycleGraphNode> get_nodes() const {
```

Comment:
use `const&` to avoid copies
```suggestion
  const std::unordered_map<std::string, LifecycleGraphNode>& get_nodes() const {
```